app/assets/javascripts/discourse/app/controllers/full-page-search.js

```diff
@@ -510,6 +511,9 @@ export default Controller.extend({
           ?.removeAttribute("open");
       }
       this.set("page", 1);
+
+      this.appEvents.trigger("full-page-search:trigger-search");
```

Comment:
Not a problem, but I'm wondering why did you add a front-end hook instead of one in the backend :thinking:

Reply:
We already had an endpoint for semantic search, so I decided to keep it instead of extending core in a different way.

Btw, this is the only way I found to trigger a new search when the core one happens. I couldn't make it work with the did-update modifier.